### ISSUE TITLE
Editor problems(textAngular)

### ISSUE THREAD

**Issue by let4be** [lavab/web#805]
I'm currently collecting a list of issues/weird behavior we want to get rid of
Feel free to post noticed issues with editor here

- [] Can't resize pasted images on chrome
- [] Laggy behavior when pasting >= 800kb images, hangs if paste larger
- [] Cannot use `B` `I` `U` formatting options if nothing is selected
- [] H1 H2 H3 isn't highlighted when selected

**Comment by let4be** [comment]
What we need from new features:
- plain text mode
- html mode
- markdown?

**Comment by andreis** [comment]
Nah just plain text and html for now, we can add markdown later

**Comment by let4be** [comment]
http://codemirror.net/mode/markdown/ looks like a good choice for plain-text, html and markdown modes

**Comment by andreis** [comment]
If @vLooz allows it :smile:

**Comment by let4be** [comment]
This one also looks nice https://stackedit.io/editor
but afaik no html mode here

**Comment by piggyslasher** [comment]
Guys, this one is superb! Has angular support. Looks sleek too!
https://froala.com/wysiwyg-editor/

It is Commercial Open Source (https://github.com/froala/angular-froala)

http://froala.com/wysiwyg-editor/pricing (inexpensive compared to other commercial options)

@vLooz need's to chime in one the Markdown mode.

**Comment by let4be** [comment]
Hm, looks tasty :)
Doesn't have textAngular childish problems like Hn/B/I/U not highlighting or issues with picture resize

**Comment by andreis** [comment]
@piggyslasher Ha, that's the one my colleagues from http://www.upb.ro/en/ did! Nice! :+1:

**Comment by andreis** [comment]
They're on Github too https://github.com/froala

I'm going to try contacting them on G+

**Comment by andreis** [comment]
It's free to try at https://www.froala.com/wysiwyg-editor/pricing

We can get it via bower: https://github.com/froala/wysiwyg-editor/issues/17

Two new heavy deps (jquery and fa), but it shouldn't be a _big_ issue. Let's give this a try!

Docs: https://www.froala.com/wysiwyg-editor/docs

**Comment by let4be** [comment]
Sounds awesome :+1: 
I will check it out right after merge and deploy a bunch of fixes

**Comment by let4be** [comment]
btw @andreis their plans page looks awesome, we should do something similar :D

**Comment by andreis** [comment]
Just noticed they have framework integrations on github: https://github.com/froala/angular-froala

**Comment by let4be** [comment]
so! Finally I found a problem with integration it was font-awesome dependency and it must be loaded after our css

There is a integration preview available at: https://github.com/lavab/web/tree/feature/editor

So far my UX is very positive about this editor - it feels and works so much better than textAngular.

pros:
- it's fast, intuitive and very easy to use
- doesn't have childish problems of textAngular(B,I,U and all other buttons work properly)
- from coding point it has modular architecture(plugins) so technically it should be easy to extend/replace parts of it.
- it's a paid product and it has dedicated developer team so we save our resources(time && money) buying ready to use product that we _don't_ have to support. textAngular has just one developer working on it in a free time

cons(not even a con, just a note):
- it has jquery dependency and is a heavy thing, around css+js ~ 400kb, but just ~120 kb under gzip which is fair trade-off for such complex thing as a decent editor
On the other hand jquery may be useful in internal Lavaboom development, sometimes there is a need to do some more/less complex DOM request from angular.js directive and JQ is an excellent tool to save your time here.

I'm all for this editor

**Comment by let4be** [comment]
But before I'd really like to ask a few question form devs, so if possible connect us somehow, @andreis 
short list of questions:
- how would it be possible to intercept d&d event when user drags something in the editor
- initialized event seems to be broken

**Comment by andreis** [comment]
I asked for help

**Comment by 3nsoft** [comment]
Froala does look nice, but, a *huge BUT*, if you insert it into UI, paying/negotiating a license for your single site, no one else will be able to run Lavaboom on their machine, as it will no longer be clean IP-wise. Froala is not MIT, is it?

Wasn't that your promise, that people will be able to run Lavaboom instances, in docks, etc.? Or, is such understanding incorrect?

Yes, Froala tech is nice. And imitation is the best compliment. :+1: 
Have you checked this: https://github.com/Voog/wysihtml may be it can be angularized?

**Comment by let4be** [comment]
I agree with you regarding froala, their licensing may be a no-go in this case
@andreis ?

https://github.com/Voog/wysihtml looks promising
I will check it out
